V2G_Libraries/v2gLib/src/Decoder.cpp

```diff
@@ -52,53 +52,63 @@ exi_bitstream_t Decoder::init_exi_bitstream(const uint8_t *buffer, size_t len)
 
 template <typename T>
 int Decoder::decode_exi(const uint8_t *buffer,
-                        size_t len, char *xmlOut,
-                        int (*decodeFunction)(exi_bitstream_t *, T *, char *))
+                        size_t len,
+                        char *xmlOut,
+                        size_t xmlOut_size,
+                        int (*decodeFunction)(exi_bitstream_t *, T *, char *, size_t))
 {
     T exi;
     exi_bitstream_t input_stream = init_exi_bitstream(buffer, len);
-    return decodeFunction(&input_stream, &exi, xmlOut);
+    return decodeFunction(&input_stream, &exi, xmlOut, xmlOut_size);
 }
 
 result_decode Decoder::decode_message(const std::string &exiIn, const std::string &schemaIn)
 {
     result_decode result;
 
     const uint8_t *buffer = hex_str_to_uint8(exiIn.c_str());
-    size_t len = strlen(exiIn.c_str()) / 2;
-    xml_buffer[0] = 0; // reset buffer
+    xml_buffer[0] = 0;  // reset buffer
+
+    if (exiIn.size() % 2 != 0)
```

Comment:
Shouldn't it be this way?
```suggestion
    if (!buffer)
```
The check whether exiIn has an even size is already performed in hex_str_to_uint8. And hex_str_to_uint8 can go wrong in other ways, we should check here, too.